### ISSUE TITLE
"Extension Methods" lists generic methods whose type arguments or constraints do not match

### ISSUE THREAD

**Issue by KalleOlaviNiemitalo** [dotnet/dotnet-api-docs#1540]
The `System.Net.Http.MultipartContent` class implements `IEnumerable&lt;HttpContent&gt;`. The documentation of this class lists many extension methods that are applicable to some `IEnumerable&lt;T&gt;` but not to `IEnumerable&lt;HttpContent&gt;`. Those extension methods should not be listed here because they cannot be used on `MultipartContent`. For example:

- `CopyToDataTable&lt;T&gt; (this IEnumerable&lt;T&gt; source) where T : DataRow`, because `HttpContent` is not derived from `DataRow`
- `Average (this IEnumerable&lt;int&gt; source)`, because `MultipartContent` does not implement `IEnumerable&lt;int&gt;`
- `Ancestors&lt;T&gt; (this IEnumerable&lt;T&gt; source) where T : XNode`, because `HttpContent` is not derived from `XNode`

Listing those inapplicable extension methods makes it harder to find the methods that actually can be used.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1c23630f-1e62-f2ed-3697-ed01b305d96b
* Version Independent ID: ac5fbfde-d2b1-945d-fd7b-42fb30566209
* Content: [MultipartContent Class (System.Net.Http)](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.multipartcontent?view=netframework-4.7.2#extension-methods)
* Content Source: [xml/System.Net.Http/MultipartContent.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Net.Http/MultipartContent.xml)
* Product: **dotnet-api**
* GitHub Login: @karelz
* Microsoft Alias: **ncldev**

**Comment by KalleOlaviNiemitalo** [comment]
It seems `Average (this IEnumerable<int> source)` is no longer listed as an extension method of `System.Net.Http.MultipartContent`. That is an improvement, but the methods of `System.Data.DataTableExtensions` and `System.Xml.Linq.Extensions` are still listed.

The problem also occurs at [the list of extension methods for `System.String`](https://docs.microsoft.com/en-us/dotnet/api/system.string?view=netframework-4.8#extension-methods), which shows e.g. [`CopyToDataTable<T>(IEnumerable<T>`)](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatableextensions.copytodatatable?view=netframework-4.8#System_Data_DataTableExtensions_CopyToDataTable__1_System_Collections_Generic_IEnumerable___0__). That method has the constraint `where T : System.Data.DataRow`; the `System.String` class implements `IEnumerable<char>`, which cannot satisfy the constraint.

**Comment by rzikm** [comment]
This issue does not seem to be specific to networking. @dotnet/docs any idea how to solve this issue? The extension method list seems to be generated automatically somewhere.

**Comment by gewarren** [comment]
@rzikm Yes we have an internal work item to track this: https://dev.azure.com/ceapex/Engineering/_queries/edit/347770.
